### PR TITLE
Use charrua-core 0.3.

### LINKED ISSUE #21: hostnet's use of dhcp-server should not use parse
The DHCP server/library used by hostnet, charrua-core, has [an exposed configuration type](https://github.com/haesbaert/charrua-core/blob/master/lib/dhcp_server.mli#L38).  `hostnet` could construct the configuration directly, rather than constructing a string and then requesting the DHCP server attempt to parse it into a configuration.  This would likely help in implementation of #18 .

### PR DESCRIPTION
**Pull request by yomimono**
Fixes #21 and #18.  Removes a pin, addressing #27 .

Follow-on of #46 .

### CODE REVIEW

**Comment by yomimono** [comment]
Human-tested, robot-approved. @ijc25 , this may be too late to help you, but perhaps not?

**Comment by samoht** [comment]
Did you run `make commit` inside `opam/` to update the vendored packages?

File: src/hostnet/lib/tcpip_stack.ml [hunk 1]
```diff
@@ -29,37 +28,47 @@ let rec smallest_prefix a_ip other_ips = function
...
+    hostname = "vpnkit"; (* it's us! *)
+    hosts = [ hyperkit ];
+    default_lease_time = Int32.of_int (60 * 60 * 2); (* 2 hours, from charrua defaults *)
+    max_lease_time = Int32.of_int (60 * 60 * 24) ; (* 24 hours, from charrua defaults *)
+    ip_addr = local_ip;
+    mac_addr = server_macaddr;
+    network = prefix;
+    range = (low_ip, high_ip);
+  } in
+  { peer_ip; local_ip; prefix; server_macaddr; client_macaddr; dhcp_configuration }
```
Comment (djs55) [inline src/hostnet/lib/tcpip_stack.ml]: That's looking a lot nicer!

BTW do you know if charrua allows the dynamic range to be empty? I tried various things with the string and failed to make it work.

Comment (yomimono) [inline src/hostnet/lib/tcpip_stack.ml]: Sadly, no; it explicitly disallows that case when parsing a string, and it's not expressible in the type signature of the configuration record.  I filed an issue there: see https://github.com/haesbaert/charrua-core/issues/27 .

**Comment by yomimono** [comment]
@samoht: I completely missed that there was a Makefile in there, but I think I did the equivalent manually minus the invocation of `./update-upstream.sh`.

**Comment by djs55** [comment]
@samoht is it ok to merge this as-is and then use `make commit` next time? Or will that break something?

**Comment by samoht** [comment]
as long as CI is happy it should be ok :-) I can add commit hooks to make the CI fail if you think that's useful.

**Comment by djs55** [comment]
Failing CI with a message which says "use `make commit`" would be handy!

**Pull request merged**